core/src/main/java/ai/timefold/solver/core/impl/domain/variable/listener/support/VariableListenerSupport.java

```diff
@@ -33,12 +36,26 @@ public static <Solution_> VariableListenerSupport<Solution_> create(InnerScoreDi
     private final NotifiableRegistry<Solution_> notifiableRegistry;
     private final Map<Demand<?>, SupplyWithDemandCount> supplyMap = new HashMap<>();
 
+    private final List<ListVariableEvent> listVariableEventList;
+    private final ListVariableDescriptor<Solution_> listVariableDescriptor;
+    private final List<CascadingUpdateShadowVariableDescriptor<Solution_>> cascadingUpdateShadowVarDescriptorList;
+
     private boolean notificationQueuesAreEmpty = true;
     private int nextGlobalOrder = 0;
 
     VariableListenerSupport(InnerScoreDirector<Solution_, ?> scoreDirector, NotifiableRegistry<Solution_> notifiableRegistry) {
         this.scoreDirector = scoreDirector;
         this.notifiableRegistry = notifiableRegistry;
+        this.cascadingUpdateShadowVarDescriptorList = scoreDirector.getSolutionDescriptor().getEntityDescriptors().stream()
+                .flatMap(e -> e.getDeclaredCascadingUpdateShadowVariableDescriptors().stream())
+                .toList();
+        this.listVariableDescriptor = scoreDirector.getSolutionDescriptor().getEntityDescriptors().stream()
```

Comment:
Afaik `SolutionDescriptor` has a method which can already retrieve the one and only list variable descriptor.